src/plugins/validation/oas3/semantic-validators/pagination-ibm.js

```diff
@@ -0,0 +1,192 @@
+  for (const path in resolvedSpec.paths) {
+    // For now, just consider the get operation.
+    const operation = resolvedSpec.paths[path].get;
+
+    // Skip any path that ends in a path parameter or does not have a "get" operation
+    if (/}$/.test(path) || !operation) {
+      continue;
+    }
+
+    // Find first success response code
+    const resp = Object.keys(operation.responses || {}).find(code =>
+      code.startsWith('2')
+    );
+    // Now get the json content of that response
+    const content = resp && operation.responses[resp].content;
+    const jsonResponse = content && content['application/json'];
+
+    // Can't check response schema for array property, so skip this path
+    if (
+      !jsonResponse ||
+      !jsonResponse.schema ||
+      !jsonResponse.schema.properties
+    ) {
+      continue;
+    }
+
+    // If no array at top level of response, skip this path
+    if (
+      !Object.values(jsonResponse.schema.properties).some(
+        prop => prop.type === 'array'
+      )
+    ) {
+      continue;
+    }
+
+    // Check for "limit" query param -- if none, skip this path
+    const params = operation.parameters;
+    if (
+      !params ||
+      !params.some(param => param.name === 'limit' && param.in === 'query')
+    ) {
+      continue;
+    }
+
+    // Now we know we have a "paginated list" operation, so lets perform our checks
+
+    // - The `limit` query parameter be type integer, optional, and have default and maximum values.
+
+    const limitParamIndex = params.findIndex(
+      param => param.name === 'limit' && param.in === 'query'
+    );
+    const limitParam = params[limitParamIndex];
+    if (
+      !limitParam.schema ||
+      limitParam.schema.type !== 'integer' ||
+      !!limitParam.required ||
+      !limitParam.schema.default ||
+      !limitParam.schema.maximum
+    ) {
+      result[checkStatus].push({
+        path: ['paths', path, 'get', 'parameters', limitParamIndex],
+        message:
+          'The limit parameter must be of type integer and optional with default and maximum values.'
+      });
+    }
+
+    // - If the operation has an `offset` query parameter, it must be type integer and optional
+
+    const offsetParamIndex = params.findIndex(
+      param => param.name === 'offset' && param.in === 'query'
+    );
+    if (offsetParamIndex !== -1) {
+      const offsetParam = params[offsetParamIndex];
+      if (
+        !offsetParam.schema ||
+        offsetParam.schema.type !== 'integer' ||
+        !!offsetParam.required
+      ) {
+        result[checkStatus].push({
+          path: ['paths', path, 'get', 'parameters', offsetParamIndex],
+          message: 'The offset parameter must be of type integer and optional.'
+        });
+      }
+    }
+
+    // - if the operation has a `start`, `cursor`, or `page_token` query parameter, it must be type string and optional
+
+    const startParamNames = ['start', 'cursor', 'page_token'];
```

Comment:
I'm surprised to see the flexibility here—the handbook says such a parameter MUST be called `start`.

Reply:
I suppose I'm being an easy grader here 😊